examples/daisyworld.jl

```diff
@@ -48,249 +60,269 @@ mutable struct Daisy <: AbstractAgent
     if model.scenario == :ramp
         if model.tick > 200 && model.tick <= 400
-            model.solar_luminosity += 0.005
+            model.solar_luminosity += model.solar_change
         end
         if model.tick > 500 && model.tick <= 750
-            model.solar_luminosity -= 0.0025
+            model.solar_luminosity -= model.solar_change/2
         end
+    elseif model.scenario == :change
+        model.solar_luminosity += model.solar_change
     end
 end
 
-function model_step!(model::ABM{Daisy})
-    for n in nodes(model)
-        suface_temperature!(n, model)
-        diffuse_temperature!(n, model)
-        propagate!(n, model)
+# ## Initialising Daisyworld
+
+# Here, we construct a function to initialize a Daisyworld. We use [`fill_space!`](@ref)
+# to fill the space with `Land` instances. Then, we need to know how many
+# daisies of each type to seed the planet with and what their albedo's are.
+# We also want a value for surface albedo, as well as solar intensity
+# (and we also choose between constant or time-dependent intensity with `scenario`).
+
+import StatsBase
+import DrWatson: @dict
+
+function daisyworld(;
+        griddims = (30, 30),
+        max_age = 25,
+        init_white = 0.2, # % cover of the world surface of white breed
+        init_black = 0.2, # % cover of the world surface of black breed
+        albedo_white = 0.75,
+        albedo_black = 0.25,
+        surface_albedo = 0.4,
+        solar_change = 0.005,
+        solar_luminosity = 1.0, # initial luminosity
+        scenario = :default,
+    )
+
+    space = GridSpace(griddims, moore = true, periodic = true)
+    properties = @dict max_age surface_albedo solar_luminosity solar_change scenario
+    properties[:tick] = 0
+    ## create a scheduler that only schedules Daisies
+    daisysched(model) = [a.id for a in allagents(model) if a isa Daisy]
+    model = ABM(Union{Daisy, Land}, space;
+        scheduler = daisysched, properties = properties, warn = false
+    )
+
+    ## fill model with `Land`: every grid cell has 1 land instance
+    fill_space!(Land, model, 0.0) # zero starting temperature
+
+    ## Populate with daisies: each cell has only one daisy (black or white)
+    white_nodes = StatsBase.sample(1:nv(space), Int(init_white * nv(space)); replace = false)
+    for n in white_nodes
+        wd = Daisy(nextid(model), vertex2coord(n, space), :white, rand(0:max_age), albedo_white)
+        add_agent_pos!(wd, model)
+    end
+    allowed = setdiff(1:nv(space), white_nodes)
+    black_nodes = StatsBase.sample(allowed, Int(init_black * nv(space)); replace = false)
+    for n in black_nodes
+        wd = Daisy(nextid(model), vertex2coord(n, space), :black, rand(0:max_age), albedo_black)
+        add_agent_pos!(wd, model)
     end
-    model.tick += 1
-    solar_activity!(model)
-end
 
-function agent_step!(agent::Daisy, model::ABM{Daisy})
-    agent.age += 1
-    agent.age >= model.max_age && kill_agent!(agent, model)
+    return model
 end
```

Comment:
`nothing #hide`

Reply:
Why do you add these `nothing # hide` statements?

Reply:
So that the text reads better in the docs. Don't see a reason for things like
```julia
agent_step!(agent::Land, model::DaisyWorld) = nothing
```
```julia
agent_step! (generic function with 2 methods)
```

The `nothing #hide` drops the last bit.

Reply:
The argument goes both ways, because you can understand that this code has been "run". In general I have not a strong preference, but for tutorials it might be worth it to explicitly show what is run code and what is not.

Reply:
I agree with Datseris that this feedback from running the code might be really helpful for beginners working through the tutorials.